galasa-ui/src/tests/components/AccessTokenSection.test.tsx

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import React from 'react';
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
+import AccessTokensSection from '@/components/AccessTokensSection';
+import { AuthTokens } from '@/generated/galasaapi';
+
+// Mock Carbon components
+jest.mock('@carbon/react', () => ({
+  Loading: () => <div data-testid="loading">Loading...</div>,
+  Button: ({ onClick, disabled, children, className }: any) => (
+    <button onClick={onClick} disabled={disabled} className={className}>
+      {children}
+    </button>
+  ),
+}));
+
+// Mock TokenCard to render a button that calls the passed callback
+jest.mock('@/components/tokens/TokenCard', () => {
+  const MockTokenCard = ({ token, selectTokenForDeletion }: any) => (
+    <button
+      data-testid={`token-card-${token.tokenId}`}
+      onClick={() => selectTokenForDeletion(token.tokenId)}
+    >
+      TokenCard {token.tokenId}
+    </button>
+  );
+
+  MockTokenCard.displayName = 'MockTokenCard';
+  return MockTokenCard;
+
+});
+
+// Mock TokenRequestModal to display its isDisabled state
+jest.mock('@/components/tokens/TokenRequestModal', () => {
+  const MockDeleteBtn = ({ isDisabled }: any) => (
+    <div data-testid="token-request-modal">
+      TokenRequestModal {isDisabled ? 'Disabled' : 'Enabled'}
+    </div>
+  );
+
+  MockDeleteBtn.displayName = "MockDeleteBtn";
+  return MockDeleteBtn;
+});
+
+// Mock TokenDeleteModal to display its presence
+jest.mock('@/components/tokens/TokenDeleteModal', () => {
+  const MockModal =  (props: any) => (
+    <div data-testid="token-delete-modal">TokenDeleteModal</div>
+  );
+
+  MockModal.displayName = "MockModal";
+  return MockModal;
+
+});
+
+// Mock ErrorPage to display error message
+jest.mock('@/app/error/page', () => {
+  const MockErrorPage = () => <div data-testid="error-page">Error Page</div>;
+  MockErrorPage.displayName = "MockErrorPage";
+
+  return MockErrorPage;
+});
```

Comment:
Not sure if all of these mocks are needed. The error page definitely doesn't need mocking out since we can just check the rendered text in the document.